scripts/partition/plot.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2023 MosaicML Streaming authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Plot results of comparing streaming dataset partitioning algorithms."""
+
+from argparse import ArgumentParser, Namespace
+from typing import Dict, List, Tuple
+
+from matplotlib import pyplot as plt
+
+mark_sizes = [
+    ('CIFAR-10 (50K)', 50_000, '#dfd'),
+    ('ImageNet (1.2M)', 1_281_167, '#cfc'),
+    ('C4 (365M)', 364_868_892, '#8c8'),
+    ('LAION-5B (5.85B)', 5_850_000_000, '#484'),
```

Comment:
Can this be user args ?

Reply:
I looked into packing this data into a string flag, and am thinking no, this should remain as is

- There is no use case where you would need to dynamically vary these dataset size/time comparison lines
- Horizontally, if the dataset size comparisons could be improved, you'd want to fix it permanently for all plots by editing the source, not temporarily by editing a flag
- Vertically, if the arrangement of "Instant", "Fast", and "Slow" could be improved, you'd want to fix it permanently for all plots by editing the source, not temporarily by editing a flag
- Flags values like `CIFAR-10 (50K),50000,#dfd;ImageNet (1.2M),1281167,#cfc;C4 (365M),364868892,#8c8;LAION-5B (5.85B),5850000000,#484` are not human-friendly
- Flag packing/unpacking would be half the file while not improving anything, etc